### PR TITLE
e2e test access to create/edit/delete goals

### LINKED ISSUE #1707: e2e test access to create/edit/delete goals

### PR DESCRIPTION
**Pull request by asabotovich**
## PR includes

- [x] Specs

## Related issues

Resolve #1707

### CODE REVIEW

File: src/components/GoalPage/GoalPage.tsx [hunk 1]
```diff
@@ -189,6 +189,7 @@ export const GoalPage = ({ user, ssrTime, params: { id } }: ExternalPageProps<{
                                         text={tr('Edit')}
                                         iconLeft={<IconEditOutline size="xs" />}
                                         onClick={dispatchModalEvent(ModalEvent.GoalEditModal)}
+                                        {...issueEditButton.attr}
```
Comment (awinogradov) [inline src/components/GoalPage/GoalPage.tsx:192]: goalEditButton

File: src/components/GoalSidebar/GoalSidebar.tsx [hunk 2]
```diff
@@ -179,6 +180,7 @@ export const GoalSidebar: FC<GoalSidebarProps> = ({
                             icon={<IconBinOutline size="xs" />}
                             text={tr('Archive goal')}
                             onClick={dispatchModalEvent(ModalEvent.GoalDeleteModal)}
+                            {...issueDeleteButton.attr}
```
Comment (awinogradov) [inline src/components/GoalSidebar/GoalSidebar.tsx:183]: goalDeleteButton

File: src/utils/domObjects.ts [hunk 3]
```diff
@@ -67,18 +79,13 @@ export const projectSettingsConfirmTransferProjectButton =
...
-export const goalActionCreateOneMore = goalCreateForm.add('createOneMore');
-export const goalActionCreateOnly = goalCreateForm.add('createOnly');
-export const goalSaveButton = goalCreateForm.add('saveButton');
-export const goalCancelButton = goalCreateForm.add('cancelButton');
 
 export const issuePage = DO('issuePage');
 export const issuePageHeader = issuePage.add('header');
 export const issuePageHeaderTitle = issuePageHeader.add('title');
 export const issuePageHeaderParent = issuePageHeader.add('parent');
+export const issueEditButton = issuePage.add('edit');
```
Comment (awinogradov) [inline src/utils/domObjects.ts:87]: lets rename all of them with `goal*`. Thanks!

File: cypress/plugins/db.ts [hunk 4]
```diff
@@ -90,6 +90,11 @@ export const initDb = (on: DbPluginEvents) => {
         },
 
         'db:remove:goal': async ({ id }) => {
+            await prisma.goalHistory.deleteMany({
```
Comment (awinogradov) [inline cypress/plugins/db.ts:93]: I'm not sure is ok. It must works with cascade

File: cypress/support/index.ts [hunk 5]
```diff
@@ -52,4 +76,75 @@ Cypress.Commands.addAll({
...
+        cy.get(goalForm.query).should('exist').and('be.visible');
+        cy.get(goalTitleInput.query).type(fields.title);
+        cy.get(goalDescriptionInput.query).type(fields.description);
+        cy.get(projectsCombobox.query).click();
+        cy.get(comboboxInput.query).type(fields.projectTitle.slice(0, 4));
+        cy.get(comboboxItem.query).click();
+
+        cy.get(projectsCombobox.query).contains(keyPredictor(fields.projectTitle));
+
+        cy.get(goalActionCreateOnly.query).should('exist').and('be.visible').click();
```
Comment (awinogradov) [inline cypress/support/index.ts:93]: What plan u have to check other ways?

File: cypress/support/index.ts [hunk 6]
```diff
@@ -22,6 +35,17 @@ type CreateProjectFields = {
     description: string;
 };
 
+type CreateGoalFields = {
```
Comment (awinogradov) [inline cypress/support/index.ts:38]: Copy/paste

**Review by awinogradov — CHANGES_REQUESTED** [review]

File: prisma/schema.prisma [hunk 7]
```diff
@@ -355,9 +355,9 @@ model GoalHistory {
 
 model GoalAchieveCriteria {
   id               String   @id @default(cuid())
-  goal             Goal     @relation("GoalCriterion", fields: [goalId], references: [id])
+  goal             Goal     @relation("GoalCriterion", fields: [goalId], references: [id], onDelete: Cascade)
   goalId           String
-  goalAsCriteria   Goal?    @relation("GoalAsCriteria", fields: [goalIdAsCriteria], references: [id])
+  goalAsCriteria   Goal?    @relation("GoalAsCriteria", fields: [goalIdAsCriteria], references: [id], onDelete: Cascade)
```
Comment (awinogradov) [inline prisma/schema.prisma:360]: think twice

**Review by awinogradov — APPROVED** [review]

**Review by 9teen90nine — APPROVED** [review]

**Pull request merged**